### PR TITLE
[CIS-1304, CIS-1305, CIS-1306, CIS-1307] Verify XCFrameworks integration

### PR DESCRIPTION
**Pull request by polqf**
### 🔗 Issue Link
https://stream-io.atlassian.net/browse/CIS-1304
https://stream-io.atlassian.net/browse/CIS-1305
https://stream-io.atlassian.net/browse/CIS-1306
https://stream-io.atlassian.net/browse/CIS-1307

### 🎯 Goal

Verify integration of XCFrameworks using:
- Manual import
- Carthage
- SwiftPM
- CocoaPods

Kept OSS approach for all the integration methods. Info below 👇 

### 🛠 Implementation

Find all the info here:
https://www.notion.so/stream-chat-fe-sdk/Module-stability-7aaa6f4f641647f1850008acc17a7227#4ebc3c2df9b84588864b6d996e40491d

### 🧪 Testing

Integrated using both OSS and XCFramework approaches for all the listed integration methods

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Affected documentation updated (docusaurus, tutorial, CMS (task created)

### CODE REVIEW

File: StreamChatUI.podspec [hunk 1]
```diff
@@ -1,25 +1,23 @@
...
-    spec.requires_arc = true
+  spec.homepage = "https://getstream.io/chat/"
+  spec.license = { :type => "BSD-3", :file => "LICENSE" }
+  spec.author = { "getstream.io" => "support@getstream.io" }
+  spec.social_media_url = "https://getstream.io"
 
-    spec.source_files = "Sources/StreamChatUI/**/*.swift"
-    spec.exclude_files = ["Sources/StreamChatUI/**/*_Tests.swift", "Sources/StreamChatUI/**/*_Mock.swift"]
-    spec.resource_bundles = { "StreamChatUIResources" => ["Sources/StreamChatUI/Resources/**/*"] }
+  spec.swift_version = "5.2"
```
Comment (bielikb) [inline StreamChatUI.podspec:12]: Our CI uses Swift 5.5 atm. Dunno if we're supporting also Swift version 5.2

```suggestion
  spec.swift_version = "5.5"
```

**Comment by bielikb** [comment]
@polqf the `binary podspecs` should require newer versions of the cocoapods to be used - otherwise the integrators might hit some bugs that were apparent in older versions.

Please add this requirement to our podspecs.

```
"cocoapods_version": ">= 1.11.0"
```

File: StreamChatUI.podspec (hunk 1 above)
Comment (polqf) [reply StreamChatUI.podspec:12]: I was trying to do as little changes as possible. Maybe this is something we should first debate internally?

Comment (bielikb) [reply StreamChatUI.podspec:12]: ok I see your point ;)

**Review by bielikb — APPROVED** [review]

**Pull request merged**